sympy/geometry/point.py

```diff
@@ -1062,52 +1217,24 @@ def are_collinear(*points):
 
     @staticmethod
     def are_coplanar(*points):
-        """
-
-        This function tests whether passed points are coplanar or not.
-        It uses the fact that the triple scalar product of three vectors
-        vanishes if the vectors are coplanar. Which means that the volume
-        of the solid described by them will have to be zero for coplanarity.
-
-        Parameters
-        ==========
-
-        A set of points 3D points
-
-        Returns
-        =======
-
-        boolean
-
-        Examples
-        ========
-
-        >>> from sympy import Point3D
-        >>> p1 = Point3D(1, 2, 2)
-        >>> p2 = Point3D(2, 7, 2)
-        >>> p3 = Point3D(0, 0, 2)
-        >>> p4 = Point3D(1, 1, 2)
-        >>> Point3D.are_coplanar(p1, p2, p3, p4)
-        True
-        >>> p5 = Point3D(0, 1, 3)
-        >>> Point3D.are_coplanar(p1, p2, p3, p5)
-        False
-
-        """
-        from sympy.geometry.plane import Plane
-        points = list(set(Point.pointify(p, dimension=3) for p in points))
+        from sympy.geometry.plane import Plane, Line
+        points = [Point(i) for i in points]
+        if any(i.ambient_dimension != 3 for i in points):
+            return Point.are_coplanar(*points)  # raise error there
+        points = list(uniq(points))
+        if len(points) > 2:
+            line = Line(*points[:2])
+            # keep only the first 2 points and any others that are
+            # not collinear
+            points = line.args + tuple(p for p in points[2:] if p not in line)
         if len(points) < 3:
-            raise ValueError('At least 3 points are needed to define a plane.')
-        a, b = points[:2]
-        for i, c in enumerate(points[2:]):
-            try:
-                p = Plane(a, b, c)
-                for j in (0, 1, i):
-                    points.pop(j)
-                return all(p.is_coplanar(i) for i in points)
-            except ValueError:
-                pass
-        raise ValueError('At least 3 non-collinear points needed to define plane.')
+            return False
```

Comment:
See the updated changes. I now answer the question via `affine_rank` which is going to reject cases where there are not enough points to define the object about which a query is being made.